### ISSUE TITLE
Validation errors reported by zod are not useful

### ISSUE THREAD

**Issue by Fleny113** [discordjs/builders#64]
### Issue description

When creating a SlashCommandBuilder and doing toJSON() zod thrown about an recived undefined and expect a string.

Step to reproduce:
1. Import `SlashCommandBuilder` from `@discordjs/builders`
2. Create a new Builder and call `toJSON()`
3. zod should throw an error from `node_modules/zod/lib/index.mjs:1`

[zodError.log](https://github.com/discordjs/builders/files/7729374/zodError.log)

### Code sample

```typescript
import { SlashCommandBuilder } from "@discordjs/builders";
new SlashCommandBuilder().setName("test").toJSON();
```

### @discordjs/builders version

0.9.0

### Node.js version

node.js v17.2.0 | Typescript 4.5.4

### Operating system

Windows 11

### Priority this issue should have

High (immediate attention needed)

**Comment by vladfrangu** [comment]
Is your code..minified/bundled at all? Because it definitely throws an error, as seen at the end, but a really, REALLY not helpful one

**Comment by Khasms** [comment]
The issue is you didn't provide a description but the description is required.

**Comment by Fleny113** [comment]
> Is your code..minified/bundled at all? Because it definitely throws an error, as seen at the end, but a really, REALLY not helpful one

No, i having this issue from typescript (and tsc) but i tried directly from nodejs and still throw error

> The issue is you didn't provide a description but the description is required.

Sorry, i added it.

**Comment by vladfrangu** [comment]
That's not what @Khasms meant haha. You need to call `.setDescription` on the builder you created! For further questions, feel free to ask in our [Discord server](https://discord.gg/djs)

**Comment by Fleny113** [comment]
> That's not what @Khasms meant haha. You need to call `.setDescription` on the builder you created! For further questions, feel free to ask in our [Discord server](https://discord.gg/djs)

I known that is not he mean but really, i don't known what to write, the errore is useless.
Anyway, thanks for help.

**Comment by vladfrangu** [comment]
Hrmm, true true, the errors could be better.. I'll be reopening this but with a different title 👍

**Comment by Fleny113** [comment]
> Hrmm, true true, the errors could be better.. I'll be reopening this but with a different title 👍

I updated the description to made that more consistent with title

**Comment by hampuskraft** [comment]
Hi there, the only alternative as I can see it is to wrap parsing with a custom error that sets `this.stack = undefined`, resulting in the below:

![image](https://user-images.githubusercontent.com/24176136/146599118-2c5b6bf2-2020-4505-a7fc-21d9ffb3e6d4.png)

```ts
import { z, ZodIssue } from 'zod';

export class ValidationError extends Error {
	public constructor(public readonly issues: ZodIssue[]) {
		super('A validation error has occurred.');
		this.name = 'ValidationError';
		this.stack = undefined;
	}
}

export function parseZodSchema<T extends z.ZodType<any>>(schema: T, data: unknown) {
	try {
		schema.parse(data);
	} catch (error) {
		if (error instanceof z.ZodError) {
			throw new ValidationError(error.issues);
		}
	}
}
```

If this seems like a good tradeoff to you I'm willing to PR it. If consumers would expect a validation error for some reason the `ValidationError` class can be exported so it can be caught.

Kind regards,
Hampus Kraft.

**Comment by Khasms** [comment]
I can't reproduce the code dump at the beginning of the error, it outputs as it should on my end. But, as far as the error not being helpful, I opened #61 a week ago to improve them, so give some feedback on that if you see any other messages that need to be improved.

**Comment by kyranet** [comment]
Not a good tradeoff since that makes parsing +2x slower, which adds up to how slow Zod's validation alreaady is, see https://github.com/discordjs/discord.js/pull/7067#issuecomment-989502034
